packaging_automation/update_docker.py

```diff
@@ -0,0 +1,147 @@
+    content = process_template_file(project_version, template_path,
+                                    "alpine/alpine.tmpl.dockerfile", postgres_version)
+    dest_file_name = f"{exec_path}/alpine/Dockerfile"
+    write_to_file(content, dest_file_name)
+
+
+def update_docker_file_for_postgres12(project_version: str, template_path: str, exec_path: str):
+    content = process_template_file(project_version, template_path,
+                                    "postgres-12/postgres-12.tmpl.dockerfile")
+    dest_file_name = f"{exec_path}/postgres-12/Dockerfile"
+    write_to_file(content, dest_file_name)
+
+
+def get_new_changelog_entry(project_version: str, postgres_version: str = ""):
+    header = f"### citus-docker v{project_version}.docker ({datetime.strftime(datetime.now(), '%B %d,%Y')}) ###\n"
+    citus_bump_str = f"\n* Bump Citus version to {project_version}\n"
+    postgres_bump_str = f"\n* Bump PostgreSQL version to {postgres_version}\n"
+
+    changelog_entry = f"{header}{citus_bump_str}"
+    if postgres_version:
+        changelog_entry = f"{changelog_entry}{postgres_bump_str}"
+    changelog_entry = f"{changelog_entry}\n"
+    return changelog_entry
+
+
+def update_changelog(project_version: str, exec_path: str, postgres_version: str = ""):
+    latest_changelog = get_new_changelog_entry(project_version, postgres_version)
+    changelog_file_path = f"{exec_path}/CHANGELOG.md"
+    with open(changelog_file_path, "r+") as reader:
+        if not (f"({project_version}" in reader.readline()):
+            reader.seek(0, 0)
+            old_changelog = reader.read()
+            changelog = f"{latest_changelog}{old_changelog}"
+            reader.seek(0, 0)
+            reader.write(changelog)
+        else:
+            raise ValueError("Already version in the changelog")
+
+
+def update_all_docker_files(project_version: str, tools_path: str, exec_path: str, postgres_version: str):
+    template_path = f"{tools_path}/packaging_automation/templates/docker"
+    pkgvars_file = f"{exec_path}/pkgvars"
+
+    if postgres_version:
+        update_pkgvars(project_version, template_path, pkgvars_file, postgres_version)
+
+    persisted_postgres_version = read_postgres_version(pkgvars_file)
+
+    update_docker_file_for_latest_postgres(project_version, template_path, exec_path, persisted_postgres_version)
+    update_regular_docker_compose_file(project_version, template_path, exec_path)
+    update_docker_file_alpine(project_version, template_path, exec_path, persisted_postgres_version)
+    update_docker_file_for_postgres12(project_version, template_path, exec_path)
+    update_changelog(project_version, exec_path, postgres_version)
+
+
+def read_postgres_version(pkgvars_file: str) -> str:
+    if os.path.exists(pkgvars_file):
+        with open(pkgvars_file, "r") as reader:
+            content = reader.read()
+            lines = content.splitlines()
+            for line in lines:
```

Comment:
I guess we are already reading a single-line file here.
So instead of `for line in lines:`, maybe we can just do `reader.readline()`.

Reply:
I used reader.readlines because I think it is more readable. If I used readline I need to use while loop and check if it is the end of the file. If it is ok for you, I prefer this one

Reply:
why would we need a while/for loop ? I guess we only need to read a single line from that file, isn't it correct ?

Reply:
Bu this single line can be anywhere in the file so I need to check all the lines until I find the exact line